Assets/SEE/XR/RadialSelection.cs

```diff
@@ -0,0 +1,409 @@
+using SEE.Controls.Actions;
+using SEE.GO;
+using System.Collections.Generic;
+using System.Linq;
+using TMPro;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.InputSystem;
+using UnityEngine.UI;
+using UnityEngine.XR.Interaction.Toolkit;
+using SEE.Utils;
+
+/// <summary>
+/// This class is used, to open a
+/// radial menu in VR.
+/// This script is based on this tutorial: https://www.youtube.com/watch?v=n-xPN1v3dvA
+/// </summary>
+public class RadialSelection : MonoBehaviour
+{
+    /// <summary>
+    /// The number of radial-parts.
+    /// </summary>
+    int numberOfRadialPart = 0;
+    /// <summary>
+    /// The prefab for the single radial-parts.
+    /// </summary>
+    public GameObject radialPartPrefab;
+    /// <summary>
+    /// The canvas on which the radial should be shown.
+    /// </summary>
+    public Transform radialPartCanvas;
+    /// <summary>
+    /// The angle between the radial-parts.
+    /// </summary>
+    public float angleBetweenPart = 10;
+    /// <summary>
+    /// The radial-parts which got spawned.
+    /// </summary>
+    private List<GameObject> spawnedParts = new List<GameObject>();
+    /// <summary>
+    /// The transform of the controller.
+    /// </summary>
+    public Transform handTransform;
+    /// <summary>
+    /// The currently selected radial-part.
+    /// </summary>
+    private int currentSelectedRadialPart = -1;
+    /// <summary>
+    /// The selected radial-part.
+    /// </summary>
+    public UnityEvent<int> OnPartSelected;
+    /// <summary>
+    /// The button, which triggers the radial-menu.
+    /// </summary>
+    public InputActionReference radialMenu;
+    /// <summary>
+    /// All actions, which are currently available.
+    /// </summary>
+    List<string> actions = new();
+    /// <summary>
+    /// Alle submenus and their entries.
+    /// </summary>
+    List<(string, List<string>)> subMenus = new List<(string, List<string>)>();
```

Comment:
Wouldn't a dictionary be better suited here?

Reply:
Because we access this data structure by index with a given i, I don't think it would make much of a difference here. I found it easier to implement this way.